app/feature/feature-addon-purchase/src/main/kotlin/com/hedvig/android/feature/addon/purchase/ui/triage/TravelAddonTriageDestination.kt

```diff
@@ -0,0 +1,165 @@
+import androidx.compose.foundation.layout.fillMaxSize
+import androidx.compose.foundation.layout.fillMaxWidth
+import androidx.compose.foundation.layout.height
+import androidx.compose.foundation.layout.only
+import androidx.compose.foundation.layout.padding
+import androidx.compose.foundation.layout.safeDrawing
+import androidx.compose.foundation.layout.windowInsetsPadding
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.LaunchedEffect
+import androidx.compose.runtime.getValue
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.tooling.preview.PreviewParameter
+import androidx.compose.ui.tooling.preview.datasource.CollectionPreviewParameterProvider
+import androidx.compose.ui.unit.dp
+import androidx.lifecycle.compose.collectAsStateWithLifecycle
+import androidx.lifecycle.compose.dropUnlessResumed
+import com.hedvig.android.design.system.hedvig.ButtonDefaults.ButtonSize.Large
+import com.hedvig.android.design.system.hedvig.HedvigErrorSection
+import com.hedvig.android.design.system.hedvig.HedvigFullScreenCenterAlignedProgress
+import com.hedvig.android.design.system.hedvig.HedvigPreview
+import com.hedvig.android.design.system.hedvig.HedvigTextButton
+import com.hedvig.android.design.system.hedvig.HedvigTheme
+import com.hedvig.android.design.system.hedvig.Surface
+import com.hedvig.android.feature.addon.purchase.ui.triage.TravelAddonTriageState.Failure
+import com.hedvig.android.feature.addon.purchase.ui.triage.TravelAddonTriageState.Loading
+import com.hedvig.android.feature.addon.purchase.ui.triage.TravelAddonTriageState.Success
+import hedvig.resources.R
+
+@Composable
+internal fun TravelAddonTriageDestination(
+    viewModel: TravelAddonTriageViewModel,
+    navigateUp: () -> Unit,
+    launchFlow: (insuranceIds: List<String>) -> Unit,
+    onNavigateToNewConversation: () -> Unit,
+) {
+  val uiState: TravelAddonTriageState by viewModel.uiState.collectAsStateWithLifecycle()
+  StartChangeTierFlowScreen(
+    uiState = uiState,
+    navigateUp = navigateUp,
+    reload = {
+      viewModel.emit(TravelAddonTriageEvent.Reload)
+    },
+    launchFlow = launchFlow,
+    onNavigateToNewConversation = onNavigateToNewConversation,
+  )
+}
+
+@Composable
+private fun StartChangeTierFlowScreen(
+    uiState: TravelAddonTriageState,
+    navigateUp: () -> Unit,
+    reload: () -> Unit,
+    launchFlow: (List<String>) -> Unit,
+    onNavigateToNewConversation: () -> Unit,
+) {
+  when (uiState) {
+    is Failure -> {
+      FailureScreen(
+        reload = reload,
+        popBackStack = navigateUp,
+        reason = uiState.reason,
+        onNavigateToNewConversation = onNavigateToNewConversation,
+      )
+    }
+
+    Loading -> HedvigFullScreenCenterAlignedProgress()
+
+    is Success -> {
+      LaunchedEffect(uiState.insuranceIds) {
+        val params = uiState.insuranceIds
+        launchFlow(params)
+      }
+    }
```

Comment:
In the success case here we'd be emitting no UI at all.
Without knowing how this looks like, often this has led to animation glitches in navigation since we go from taking up 0x0 size to taking up the full size, and this size change is done with an animation by expanding the content from the center or top-left which is not what we want.

Can we either add an empty Box() in here, or even better wrap the entire `when` statement here in a `Surface(color = ...background, modifier = Modifier.fillMaxSize()) { when(uiState)... }` so that we always do emit at least the background properly?